src/frontend/core/scheduler/JobScheduler.cpp

```diff
@@ -40,27 +40,68 @@ void *startScheduler(void *dummyPt) {
             JobRequest request = jobQueue.top();
             int priority = request.priority;
-            std::string masterIP = request.getMasterIP();
 
             if (priority == Conts::HIGH_PRIORITY_DEFAULT_VALUE) {
                 string perfCategory = request.getParameter(Conts::PARAM_KEYS::CATEGORY);
                 string command = request.getJobType();
                 string graphId = request.getParameter(Conts::PARAM_KEYS::GRAPH_ID);
-                bool isResourcesSufficient = performanceUtil.isResourcesSufficient(graphId,command,perfCategory,masterIP);
-
-                if (!isResourcesSufficient) {
-                    std::this_thread::sleep_for(std::chrono::seconds(1));
-                }
             }
 
             jobQueue.pop();
             JobScheduler::processJob(request, refToScheduler->sqlite, refToScheduler->perfSqlite);
-            jobScheduler_Logger.log("##JOB SCHEDULER## Scheduled the job", "info");
+            jobScheduler_Logger.log("##JOB SCHEDULER## Scheduled the job", "info");*/
+
+            std::vector<JobRequest> pendingHPJobList;
+            std::vector<std::string> highPriorityGraphList;
+
+            while (!jobQueue.empty()) {
+                JobRequest request = jobQueue.top();
+
+                if (request.getPriority() == Conts::HIGH_PRIORITY_DEFAULT_VALUE && request.getJobType() == TRIANGLES) {
+                    pendingHPJobList.push_back(request);
+                    highPriorityGraphList.push_back(request.getParameter(Conts::PARAM_KEYS::GRAPH_ID));
+                    jobQueue.pop();
+                } else {
+                    jobQueue.pop();
+                    JobScheduler::processJob(request, refToScheduler->sqlite, refToScheduler->perfSqlite);
+                }
+            }
+
+
+            if (pendingHPJobList.size() > 0) {
+                jobScheduler_Logger.log("##JOB SCHEDULER## High Priority Jobs in Queue: " + std::to_string(pendingHPJobList.size()), "info");
+                std::string masterIP = pendingHPJobList[0].getMasterIP();
+                std::string jobType = pendingHPJobList[0].getJobType();
+                std::string category = pendingHPJobList[0].getParameter(Conts::PARAM_KEYS::CATEGORY);
+
+                std::vector<long> scheduleTimeVector = performanceUtil.getResourceAvailableTime(highPriorityGraphList,
+                        jobType, category, masterIP, pendingHPJobList);
+
+                for (int index = 0; index != pendingHPJobList.size(); ++index) {
+                    JobRequest hpRequest = pendingHPJobList[index];
+                    long queueTime = scheduleTimeVector[index];
+
+                    if (queueTime  < 0) {
+                        JobResponse failedJobResponse;
+                        failedJobResponse.setJobId(hpRequest.getJobId());
+                        failedJobResponse.addParameter(Conts::PARAM_KEYS::ERROR_MESSAGE, "Rejecting the graph because "
```

Comment:
It is not the graph that we are rejecting. It is the query(i.e., triangle count) request that we reject.

Reply:
Changed the rejection message

Reply:
Ok